osquery/tables/cloud/azure/azure_instance_metadata.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <boost/property_tree/json_parser.hpp>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/utils/azure/azure_util.h"
+
+namespace osquery {
+namespace tables {
+namespace pt = boost::property_tree;
+
+// Azure VM IDs are unique and don't change on an instance,
+// so we can safely cache them.
+static std::string kCachedVmId;
```

Comment:
Why do we need to cache this value? If you are always parsing it from the meta-data lookup then we might as well not waste persistent heap storing it?

Reply:
This was originally a client suggestion: Azure's metadata service has apparently been historically unreliable, and so the thought here was that we could at minimum return our instance's VM ID even when other data couldn't be retrieved. But I agree that it's not very useful without the other metadata, so I'll drop the caching.